letsencrypt/boulder
# Run Travis tests in Docker

- Create a base Docker image containing the tools we normally download as part of a Travis run. This cuts build times a lot, mainly because we don't need to build protoc from source.
- Set necessary environment variables
- Switch to MariaDB 10.1, including necessary changes to DB creation scripts.
- Tweak reloader_test so it passes when run as root on Docker
- Use prebuilt letsencrypt package from jessie-backports, instead of building the most recent one. Note that we still have to clone the certbot repo to get the test scripts, but we don't have to build and install dependencies like augeas. Note that jessie-backports does not yet have `certbot`. Once it does, we'll switch to that.
- ct-test-srv and mail-test-srv listen on 0.0.0.0
- Use prebuilt rabbitmq-setup in entrypoint.sh, which greatly speeds up repeated `docker run` commands.

Fixes #1335
